modules/administration-guide/partials/proc_collecting-dev-workspace-operator-metrics-with-prometheus.adoc

```diff
@@ -0,0 +1,60 @@
+[id="proc_collecting-dev-workspace-operator-metrics-with-prometheus_{context}"]
+= Collecting {devworkspace} operator metrics with Prometheus
+
+[role="_abstract"]
+This section describes how to use the Prometheus to collect, store, and query metrics about the {devworkspace} operator.
+
+.Prerequisites
+
+* The link:https://github.com/devfile/devworkspace-operator/blob/v0.10.0/deploy/deployment/kubernetes/objects/devworkspace-controller-metrics.Service.yaml[`devworkspace-controller-metrics` service] is exposing metrics on port `8443`.
+
+* The `devworkspace-webhookserver` service is exposing metrics on port `9443`. This is the default configuration.
+
+* Prometheus 2.26.0 or later is running and the Prometheus console is running on port `9090` with a corresponding *service* and *route*. See link:https://prometheus.io/docs/introduction/first_steps/[First steps with Prometheus].
+
+.Procedure
+
+. Create a `ClusterRoleBinding` to bind the `ServiceAccount` associated with Prometheus to the link:https://github.com/devfile/devworkspace-operator/blob/main/deploy/deployment/kubernetes/objects/devworkspace-controller-metrics-reader.ClusterRole.yaml[devworkspace-controller-metrics-reader] `ClusterRole`.
+This step is required to access {devworkspace} metrics because access is protected with role-based access control (RBAC).
++
+.ClusterRole example
+====
+include::example$metrics-reader-cluster-role.adoc[]
+====
+
++
+.ClusterRoleBinding example
+====
+include::example$metrics-reader-cluster-role-binding.adoc[]
+====
+
+. Configure Prometheus to scrape metrics from the `8443` port exposed by the `devworkspace-controller-metrics` service and `9443` port exposed by the `devworkspace-webhookserver` service.
++
+.Prometheus configuration example
+====
+include::example$prometheus-dev-workspace-operator-config.adoc[]
```

Comment:
My latest push rebases, renames "DevWorkspace engine" to "DevWorkspace operator", and removes `WorkspaceEngineFailure`.

Also, when building, there's an error on this line:
```
35:47  error       Did you really mean 'ato'?      CheDocs.Spelling 
```
This is part of the file name, how can I fix this?

Reply:
@dkwon17, questions:

1. Is there a particular reason why you opted for these blocks to be saved as separate example files? Can you include the content of all these `include`d files in this procedure module directly?

2. Is there an upstream/downstream difference, or none? Yes/No

Reply:
Hi @max-cx ,

1. Sounds good, I will do that. There's no specific reason why these blocks were saved as separate files.
2. The examples files have no upstream/downstream difference.

Reply:
@dkwon17, as far as I am aware, the examples directory is used to store snips with alternative upstream/downstream content. In my personal opinion, it's easier to just have all those blocks right inside the procedure module file for various reasons (things like the <1> reference syntax and ease of making suggested changes during review). Don't worry about resultant file length. Blocks can be either made generic using [the syntax for user-replaced values](https://redhat-documentation.github.io/supplementary-style-guide/#user-replaced-values) (if it's something that the user has to enter); or, blocks can remain examples within the procedure file. So there is no need to create separate example files unless there is no way around it.

@themr0c, I hope I provided correct info above, if not, please correct me here.